virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -302,7 +402,7 @@ def _setup(
         self._setup_grid_neighbours()
         """Determine grid square adjacency."""
         self.functional_groups = functional_groups
-        self.model_constants = model_constants
+        self.model_constants = self.model_constants
```

Comment:
This does not make much sense. `self.model_constants` is already `self.model_constants`! I believe you can just remove it.

```suggestion
```